cachi2/core/package_managers/pip.py

```diff
@@ -204,20 +200,34 @@ def _get_pip_metadata(package_dir: RootedPath) -> tuple[str, str]:
         name = name or setup_cfg.get_name()
         version = version or setup_cfg.get_version()
 
-    if name:
-        log.info("Resolved package name: %r", name)
+    if not name:
+        log.info("Processing metadata from git repository")
+        try:
+            repo_path = get_repo_id(package_dir.root).parsed_origin_url.path.removesuffix(".git")
+            repo_name = Path(repo_path).name
+            package_subpath = package_dir.subpath_from_root
+
+            resolved_path = Path(repo_name).joinpath(package_subpath)
+            normalized_path = pkg_resources.safe_name(str(resolved_path))
+            name = normalized_path.strip("-.")
+        except UnsupportedFeature:
+            raise PackageRejected(
+                reason="Could not take name from the repository origin url",
+                solution=(
+                    """
+                    Please specify package metadata in a way that Cachi2 understands (see the docs)
+                    or make sure that the directory Cachi2 is processing is a git repository with
+                    an 'origin' remote in which case Cachi2 will infer the package name from the remote url.
+                    """
+                ),
+                docs=PIP_METADATA_DOC,
```

Comment:
This looks a bit off in the actual CLI output
```
2023-07-25 16:17:46,838 ERROR PackageRejected: Could not take name from the repository origin url
Error: PackageRejected: Could not take name from the repository origin url

                      Please specify package metadata in a way that Cachi2 understands (see the docs)
                      or make sure that the directory Cachi2 is processing is a git repository with
                      an 'origin' remote in which case Cachi2 will infer the package name from the remote url.
                    
  Docs: https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#project-metadata
```
I think you either need to [`dedent()`](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) the solution and remove the newlines from both ends, or not use triple quotes (if you want to go the `dedent` route, it would be good to implement it generically in errors.py)